### PR TITLE
Update build scripts based on latest vue-cli webpack template

### PR DESCRIPTION
**Pull request by omidkrad**
I did not add any code or manually modify anything. I just created a new vue-cli project using the latest webpack template and replaced build files in vue-paper-dashboard with the ones from the new scaffolded project. I tested and it seems to be working, but please verify again if you had added any packages on top of the default webpack template make sure they are there. I also did not update any test files, I think they need to be updated too. You may want to redo it to make sure everything is in sync with the latest webpack template, in that case just feel free to reject this PR and add yours. Thanks!

### CODE REVIEW

File: build/webpack.base.conf.js [hunk 1]
```diff
@@ -19,29 +18,25 @@ var webpackConfig = {
...
-    modules: [
-      resolve('src'),
-      resolve('node_modules')
-    ],
     alias: {
-      'vue$': 'vue/dist/vue.common.js',
-      'src': resolve('src'),
-      'assets': resolve('src/assets'),
-      'components': resolve('src/components')
+      'vue$': 'vue/dist/vue.esm.js',
```
Comment (cristijora) [inline build/webpack.base.conf.js]: `src` alias should be here

File: build/webpack.base.conf.js [hunk 2]
```diff
@@ -54,35 +49,27 @@ var webpackConfig = {
...
+        options: {
           limit: 10000,
           name: utils.assetsPath('fonts/[name].[hash:7].[ext]')
         }
       }
     ]
   }
 }
-
-var esLintRule = {
```
Comment (cristijora) [inline build/webpack.base.conf.js]: There was a custom esLint config with "Enable/Disable" option in package.json which got deleted

File: config/index.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 var path = require('path')
+
 module.exports = {
   build: {
     env: require('./prod.env'),
-    index:  path.resolve(__dirname, '../dist/index.html'),
+    index: path.resolve(__dirname, '../dist/index.html'),
     assetsRoot: path.resolve(__dirname, '../dist'),
     assetsSubDirectory: 'static',
-    assetsPublicPath: '',
```
Comment (cristijora) [inline config/index.js]: assetsPublicPath should be let to `` empty so it's possible to deploy the bundle as a static website

File: package.json [hunk 4]
```diff
@@ -5,88 +5,93 @@
   "author": "cristian.jora <cristian.jora@corebuild.eu>",
   "private": true,
   "scripts": {
-    "dev": "cross-env ENABLE_ESLINT=true node build/dev-server.js",
-    "build": "cross-env ENABLE_ESLINT=true node build/build.js",
+    "dev": "node build/dev-server.js",
```
Comment (cristijora) [inline package.json]: would prefer the old build script with `ENABLE_ESLINT` option available

File: package.json [hunk 5]
```diff
@@ -5,88 +5,93 @@
...
+    "start": "node build/dev-server.js",
+    "build": "node build/build.js",
     "unit": "cross-env BABEL_ENV=test karma start test/unit/karma.conf.js --single-run",
-    "unit:watch": "cross-env BABEL_ENV=test karma start test/unit/karma.conf.js --watch",
```
Comment (cristijora) [inline package.json]: This one was handy as well for continuous unit testing

File: package.json [hunk 6]
```diff
@@ -5,88 +5,93 @@
...
+    "karma-phantomjs-shim": "^1.4.0",
+    "karma-sinon-chai": "^1.3.1",
     "karma-sourcemap-loader": "^0.3.7",
-    "karma-spec-reporter": "0.0.26",
+    "karma-spec-reporter": "0.0.31",
     "karma-webpack": "^2.0.2",
-    "lolex": "^1.5.2",
     "mocha": "^3.2.0",
-    "nightwatch": "^0.9.12",
-    "node-sass": "^4.5.0",
```
Comment (cristijora) [inline package.json]: node-sass dep got removed

**Comment by omidkrad** [comment]
@cristijora thanks for the review. I think your suggestion to update the cli project is even better, but I won't be able to do it now. It would be great if somewhere we can keep track of customizations on top of the original template, so every once in a while we can sync latest changes from the original template. I'm going to close this PR.

File: build/webpack.base.conf.js (hunk 1 above)
Comment (omidkrad) [reply build/webpack.base.conf.js]: See line 25. `src` alias is apparently changed to `'@'`. I did not make these changes, just got the latest from what `vue-cli init webpack` creates for me.

File: build/webpack.base.conf.js (hunk 2 above)
Comment (omidkrad) [reply build/webpack.base.conf.js]: I have a suggestion. If for any customizations that come on top of the original `webpack` template, we do on a separate file and then that would minimize the conflicts. For example, a file like `webpack.custom.conf.js` can be referenced from `webpack.base.conf.js` so whenever you want to sync the latest you will just replace the files and only modify them to reference custom scripts. A similar method is suggested for [customizing Bootstrap](https://bootstrapbay.com/blog/customize-bootstrap/).

**Pull request closed without merging**